travis/before_install.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+set -e
+set -o pipefail
+
+if [[ ${TRAVIS_OS_NAME} == "linux" ]]; then
+    ANDROID_SDK_VERSION="r24.0.2"
+    ANDROID_NDK_VERSION="r10d"
+    ANDROID_BUILD_TOOL_VERSION="21.1.2"
+    ANDROID_PLATFORM_VERSION="19"
+fi
+
+if [[ ${TRAVIS_OS_NAME} == "osx" ]]; then
+    brew update
+    brew tap homebrew/versions
+    brew install glfw3
+fi
+
+
+if [[ ${TRAVIS_OS_NAME} == "linux" ]]; then
+    # gcc-4.8 should be default in travis ci now (https://github.com/travis-ci/travis-cookbooks/pull/215), but for safety
+    sudo add-apt-repository ppa:ubuntu-toolchain-r/test -y
+    sudo apt-get update -qq
+    if [ "$CXX" = "g++" ]; then sudo apt-get install -qq g++-4.8; fi
+    if [ "$CXX" = "g++" ]; then export CXX="g++-4.8" CC="gcc-4.8"; fi
+    # install other required packages
+    sudo apt-get -y install git build-essential automake gdb libtool xutils-dev make cmake pkg-config libcurl4-openssl-dev
+    # not installing mesa and opengl right now
+
+    # install jdk
+    sudo apt-get -qq -y install openjdk-7-jdk ant lib32z1-dev lib32stdc++6
```

Comment:
And we also do not need to install gcc 4.8 right now, since it uses the gcc packaged with the ndk, but will need it for linux builds in future. I will comment that out also.